source/dofs/dof_renumbering.cc

```diff
@@ -1246,7 +1260,7 @@ namespace DoFRenumbering
                     const unsigned int idx = locally_owned_dof_indices.index_within_set (local_dof_indices[i]);
                     if (new_indices[idx] == numbers::invalid_dof_index)
                       {
-                        new_indices[idx] = locally_owned_dof_indices.nth_index_in_set (current_next_free_dof_index);
+                        new_indices[idx] = my_starting_index + current_next_free_dof_index;
```

Comment:
Neat, this is simpler than I expected. I would think we need to rename the variable ``current_next_free_dof_index`` though as it is now relative to the starting index. You are now returning this offset instead of the global index from this function. Is this intended?

Reply:
Yes, this relies on the observation that the *number of locally owned DoFs* is an invariant under renumbering. (I probably ought to document this somewhere.) Consequently, we can compute the starting offsets from the previous number of locally owned DoFs per process.